concent_api/core/views.py

```diff
@@ -437,5 +552,17 @@ def store_receive_out_of_band(golem_message, message_timestamp):
     receive_out_of_band_status.save()
 
 
+def decode_key(key):
+    return b64decode(key.encode('ascii'), validate = True)
+
+
 def decode_client_public_key(request):
-    return b64decode(request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY'].encode('ascii'), validate = True)
+    return decode_key(request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY'])
+
+
+def decode_requestor_public_key(request):
+    if 'HTTP_CONCENT_REQUESTOR_PUBLIC_KEY' in request.META:
+        try:
+            return decode_key(request.META['HTTP_CONCENT_REQUESTOR_PUBLIC_KEY'])
+        except TypeError:
+            return JsonResponse({'error': 'The value in the Requestor-Public-Key HTTP is not a valid base64-encoded value.'}, status = 400)
```

Comment:
You're missing the final `return`. If your intention was to to return `None`, please do it explicitly. Otherwise it looks as if you forgot about `else`.